src/Output.cpp

```diff
@@ -0,0 +1,109 @@
+#include "CodeGen_C.h"
+#include "StmtToHtml.h"
+#include "Output.h"
+#include "LLVM_Headers.h"
+#include "LLVM_Output.h"
+
+#include <fstream>
+
+namespace Halide {
+
+void output_object(const Module &module, std::string filename) {
```

Comment:
I suppose they don't need to follow the same naming convention? We could use a mix of compile_to_*, export_*, write_*, etc.

Maybe a namespace for all of these functions is in order?